omnibus/package-scripts/agent/preinst

```diff
@@ -70,18 +70,35 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
                 -c "Datadog Agent" dd-agent && \
                 { usermod -L dd-agent || echo "[ WARNING ]\tCannot lock the 'dd-agent' user account"; }
 
-        # Uninstall every datadog integration
+
         # Starting with 6.10, integrations are also uninstalled on package removal
-        # See https://github.com/DataDog/datadog-agent/pull/3066 for more details
-        PIP2_PATH=$INSTALL_DIR/embedded/bin/pip2
-        PIP3_PATH=$INSTALL_DIR/embedded/bin/pip3
-        if [ -x $PIP2_PATH ]; then
-            echo "Uninstalling Python 2 integrations..."
-            $PIP2_PATH freeze | grep ^datadog- | grep -v datadog-checks-base | xargs $PIP2_PATH uninstall -y -q --no-cache-dir || true
-        fi
-        if [ -x $PIP3_PATH ]; then
-            echo "Uninstalling Python 3 integrations..."
-            $PIP3_PATH freeze | grep ^datadog- | grep -v datadog-checks-base | xargs $PIP3_PATH uninstall -y -q --no-cache-dir || true
+
+        # Since 6.18.0, a file containing all integrations files which have been installed by
+        # the package is available. We use it to remove only the datadog-related check files which
+        # have *NOT* been installed by the package (eg: installed using the `integration` command).
+
+        if [ -f "$INSTALL_DIR/embedded/.installed_by_pkg.txt" ]; then
+            # List all files in the embedded dir of the datadog-agent install dir
```

Comment:
Let's `echo` a message here if we go through the logic? (something like `Removing integrations installed with the "agent integration" command` maybe?)